pkg/utils/managedresources/builder/resourcemanager_test.go

```diff
@@ -201,6 +203,57 @@ var _ = Describe("Resource Manager", func() {
 				Immutable: ptr.To(true),
 			}))
 		})
+
+		It("should update the secret if it exists", func() {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"foo": []byte("bar"),
+				},
+			}
+			Expect(fakeClient.Create(ctx, secret)).To(Succeed())
+
+			Expect(
+				NewSecret(fakeClient).
+					WithNamespacedName(namespace, name).
+					WithKeyValues(map[string][]byte{
+						"bar": []byte("foo"),
+					}).
+					Reconcile(ctx),
+			).To(Succeed())
+
+			Expect(fakeClient.Get(ctx, kubernetesutils.Key(namespace, name), secret)).To(Succeed())
+
+			Expect(secret).To(Equal(&corev1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Secret",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            name,
+					Namespace:       namespace,
+					ResourceVersion: "2",
+				},
+				Type: corev1.SecretTypeOpaque,
+				Data: map[string][]byte{
+					"bar": []byte("foo"),
+				},
+			}))
+		})
+
+		It("should fail to update the secret if it doesn't exist", func() {
+			secret := NewSecret(fakeClient).
+				WithNamespacedName(namespace, name).
+				WithLabels(labels).
+				CreateOnUpdate(false)
+
+			Expect(secret.Reconcile(ctx)).To(BeNotFoundError())
```

Comment:
Should this really be the expected behaviour? As a user, if I set `CreateOnUpdate=false` then I wouldn't expect a failure/error when it doesn't exist yet.

Reply:
Hm, I don't quite understand that expectation you're posting.

If you tell a client to update (only) a resource and the resource to update does not exist, then it's natural to get an error message.

Reply:
Perhaps the naming confused me already. Let's rename this field to make it more clear (in your text, you use "update only" which seems a good option) and then I'll go over it again 😉

Reply:
I tried both, but decided for `createIfNotExists` since it's appears seem more natural IMO. Hope that's fine.